common/src/main/scala/common/collections/WeightedQueue.scala

```diff
@@ -14,9 +14,9 @@ object WeightedQueue {
   * In addition to the queue, a weight function is provided that provides the weight W of an element T.
   * The total weight of the queue is accessible, as well as a method to take the head of the queue based on a max weight value.
   */
-case class WeightedQueue[T, W](innerQueue: Queue[T],
-                               private val weightFunction: T => W,
-                               weight: W)(implicit n: Numeric[W]) {
+final case class WeightedQueue[T, W](innerQueue: Queue[T],
+                                     private val weightFunction: T => W,
+                                     weight: W)(implicit n: Numeric[W]) {
```

Comment:
Oops - this is a stowaway change which doesn't have anything to do with the rest of the PR.